ReSharper.FSharp/src/FSharp.Psi.Intentions/src/Intentions/UseNamedAccessAction.fs

```diff
@@ -0,0 +1,107 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Intentions
+
+open FSharp.Compiler.Symbols
+open JetBrains.ReSharper.Feature.Services.ContextActions
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+
+[<ContextAction(Name = "UseNamedAccess", Group = "F#", Description = "Use named access inside a DU pattern")>]
+type UseNamedAccessAction(dataProvider: FSharpContextActionDataProvider) =
+    inherit FSharpContextActionBase(dataProvider)
+    
+    let mutable names = Array.empty
+    let mutable tuplePat : ITuplePat = null
+
+    override x.Text = "Use named access"
+
+    override x.IsAvailable _ =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then false else
+        // We expect a single IParenPat here
+        if pattern.Parameters.Count <> 1 then false else
+        let parenPat =
+            match pattern.Parameters.[0] with
+            | :? IParenPat as parenPat -> parenPat
+            | _ -> null
+
+        if isNull parenPat then false else
+        // This should only work when there are multiple items
+        tuplePat <- parenPat.Pattern :?> ITuplePat
+        if isNull tuplePat then false else
+        
+        let hasUnderscore =
+            tuplePat.Patterns
+            |> Seq.exists (fun p ->
+                match p with
+                | :? IWildPat -> true
+                | _ -> false)
+    
+        // We need something that is currently ignored
+        if not hasUnderscore then false else
```

Comment:
That's an interesting choice. 🙂 

Let's consider these cases:
```fsharp
match x with
| UnionType.UnionCase1(true, true, true) -> ()
| UnionType.UnionCase1(fieldName1, fieldName2, _) -> ()
```

In the first case it might make sense to enable the action, as the use of literals make it a bit harder to know what fields they match. But it's not enabled because there's no `_` patterns.

In the second case it's enabled, but I'd consider it much less useful, as the names already may say what they stand for (and pattern generation features would try to use names from the fields if possible).


The range where this action is available is also worth discussing. Normally, we try to limit such ranges because there are more and more actions being added over years, and we don't want later neither 1) to have to many at one place 2) to restrict the ranges and break habits. A union name and `_` patterns could be good ranges instead, but _maybe_ for this particular action it's worth to have a single longer range. It definitely needs some experimenting.

Reply:
Very good feedback here! I guess we should have some sort of heuristic.
If the number of wildcards and constants exceeds a certain threshold, we should propose the named access intention.

What about if 50% of the patterns are either constant or wildcards then propose the intention.
Or always if three or more constant or wildcards.
```fsharp
match x with
| A(a, _{on}) -> () // 50%
| B(a, b, _{off}) -> ()  // 33%
| C(true, true, a, b, d, c, e, _{on} ) // 60% but 3 items can be replaced
```

Reply:
We settled on: always enabling for now and improving later.